framework/source/class/qx/core/Assert.js

```diff
@@ -198,6 +198,33 @@ qx.Bootstrap.define("qx.core.Assert",
       );
     },
 
+    /**
+     * Assert that both float values are equal within the given tolerance. This
+     * might be needed because of the natural floating point inaccuracy of
+     * computers.
+     *
+     * @param expected {Float} Reference value
+     * @param found {Float} Found value
+     * @param precision {Float ? 0.01} Value by which expected and found may
+     *   differ
+     * @param msg {String} Message to be shown if the assertion fails.
+     */
+    assertEqualsFloat : function(expected, found, precision, msg)
+    {
+      this.assertNumber(expected);
+      this.assertNumber(found);
+      if (!qx.lang.Type.isNumber(precision)) {
+        precision = 0.01;
```

Comment:
this will not work well for small or very large numbers ... there are established algorithms for this kind of problem ...I can look them up for you if you wish

Reply:
http://mathjs.org/docs/datatypes/numbers.html

Equality#

Because of rounding errors in calculations, it is unsafe to compare JavaScript Numbers. For example executing 0.1 + 0.2 == 0.3 in JavaScript will return false, as the addition 0.1 + 0.2 introduces a round-off error and does not return exactly 0.3.

To solve this problem, the relational functions of math.js check whether the relative difference between the compared values is smaller than the configured option epsilon. In pseudo code (without exceptions for 0, Infinity and NaN):

diff = abs(x - y)
nearlyEqual = (diff <= max(abs(x), abs(y)) * EPSILON) OR (diff < DBL_EPSILON)
where:

EPSILON is the relative difference between x and y. Epsilon is configurable and is 1e-14 by default. See Configuration.
DBL_EPSILON is the minimum positive floating point number such that 1.0 + DBL_EPSILON != 1.0. This is a constant with a value of approximately 2.2204460492503130808472633361816e-16;
Note that the relational functions cannot be used to compare small values (< 2.22e-16). These values are all considered equal to zero.

Reply:
http://mathjs.org/docs/datatypes/numbers.html

Equality#

Because of rounding errors in calculations, it is unsafe to compare JavaScript Numbers. For example executing 0.1 + 0.2 == 0.3 in JavaScript will return false, as the addition 0.1 + 0.2 introduces a round-off error and does not return exactly 0.3.

To solve this problem, the relational functions of math.js check whether the relative difference between the compared values is smaller than the configured option epsilon. In pseudo code (without exceptions for 0, Infinity and NaN):

diff = abs(x - y)
nearlyEqual = (diff <= max(abs(x), abs(y)) * EPSILON) OR (diff < DBL_EPSILON)
where:

EPSILON is the relative difference between x and y. Epsilon is configurable and is 1e-14 by default. See Configuration.
DBL_EPSILON is the minimum positive floating point number such that 1.0 + DBL_EPSILON != 1.0. This is a constant with a value of approximately 2.2204460492503130808472633361816e-16;
Note that the relational functions cannot be used to compare small values (< 2.22e-16). These values are all considered equal to zero.

Reply:
there is even Number.EPSILON defined in Many js implementations

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/EPSILON